pkg/controllers/securitypolicy/securitypolicy_controller.go

```diff
@@ -91,7 +93,14 @@ func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	// Use once.Do to ensure gc is called only once
 	common.GcOnce(r, &once)
 
-	obj := &v1alpha1.SecurityPolicy{}
+	var obj client.Object
+	if r.Service.NSXConfig.EnableVPCNetwork {
+		obj = &crdv1alpha1.SecurityPolicy{}
+		EnableVPCNetwork = true
+	} else {
+		obj = &v1alpha1.SecurityPolicy{}
+	}
```

Comment:
Does it need to duplicate the definition of the structs?

I thought you could declare the structs in one package and reference it in another package, then you only need to deal with one struct. See how https://github.com/kubernetes/kubernetes/blob/v1.20.0/pkg/apis/extensions/register.go shares the struct with https://github.com/kubernetes/kubernetes/blob/v1.20.0/pkg/apis/apps/register.go

But not sure if `r.Client` can have some ways to differentiate which API we want to access.

Reply:
Yes, I don't find ways for `r.Client` to differentiate, so I must duplicate it.